io.openems.edge.timedata.rrd4j/src/io/openems/edge/timedata/rrd4j/RecordWorker.java

```diff
@@ -75,11 +78,10 @@ public void collectData() {
 
 		// Same second as last run? -> RRD4j can only handle one sample per second per
 		// database. Timestamps are all stored "truncated to seconds".
-		if (recordTimestamp.isEqual(this.lastRecordedTimestamp)) {
+		if (timestamp.equals(this.lastTimestamp)) {
```

Comment:
lastTimestamp is never reassigned, so this would always be false.

Reply:
Thanks for pointing out such an error again... fixed here: https://github.com/OpenEMS/openems/commit/7e758783ec9acbabefeec3a8a8d339471393369b